tools/deb/istio-iptables.sh

```diff
@@ -319,13 +326,24 @@ if [ -n "${OUTBOUND_IP_RANGES_EXCLUDE}" ]; then
   done
 fi
 
+for internalInterface in ${REDIRECT_INTERNAL_INTERFACE}; do
+    iptables -t nat -I PREROUTING 1 -i "${internalInterface}" -j RETURN
+done
+
 # Apply outbound IP inclusions.
 if [ "${OUTBOUND_IP_RANGES_INCLUDE}" == "*" ]; then
   # Wildcard specified. Redirect all remaining outbound traffic to Envoy.
   iptables -t nat -A ISTIO_OUTPUT -j ISTIO_REDIRECT
+  for internalInterface in ${REDIRECT_INTERNAL_INTERFACE}; do
+    iptables -t nat -I PREROUTING 1 -i "${internalInterface}" -j ISTIO_REDIRECT
+  done
```

Comment:
shouldn't this be an if else? When envoy sends traffic captured from pod inbound to some localhost:port, it is going to be captured by some iptable rule and end up in tap0 - which would appear as outbound traffic to the rule on line 336. From my understanding, what you seem to want here is
 if kubeVirt, capture all inbound traffic on internal_vifs and jump to istio_redirect
 else capture all outbound traffic from istio_output and jump to istio_redirect

If that is the case, for sake of clarity, isn't it easier to simply rename the REDIRECT_INTERNAL_INTERFACE to KUBEVIRT_INTERFACES ? and use an if-else in lines 337-339 ?

Reply:
please see my comment below